src/SMAIAXBackend.Application/Services/Implementations/SmartMeterCreateService.cs

```diff
@@ -39,6 +39,12 @@ public async Task<Guid> AddSmartMeterAsync(SmartMeterCreateDto smartMeterCreateD
             metadataList.Add(metadata);
         }
 
+        if (String.IsNullOrEmpty(smartMeterCreateDto.Name))
```

Comment:
You can add the annotation `[Required(AllowEmptyStrings = false)]` to the `Name` property in `SmartMeterCreateDto`. Then it should automatically check that the name is not null and not empty.